apps/web-client/src/routes/+layout.svelte

```diff
@@ -1,8 +1,38 @@
 <script lang="ts">
 	// Import main.css in order to generate tailwind classes used in the app
 	import "../main.css";
+	import { onMount } from "svelte";
+	import { pwaInfo } from "virtual:pwa-info";
+
+	onMount(async () => {
+		if (pwaInfo) {
+			console.log({ pwaInfo });
+
+			const { registerSW } = await import("virtual:pwa-register");
+			registerSW({
+				immediate: true,
+				onRegistered(r) {
+					r &&
+						setInterval(() => {
+							console.log("Checking for sw update");
+							r.update();
+						}, 20000);
+					console.log(`SW Registered: ${r}`);
+				},
+				onRegisterError(error) {
+					console.log("SW registration error", error);
```

Comment:
We should do something with this. Once toast library is added, it could potentially be communicated to client, or is it unimportant to them? What are main reasons it could fail? Should we retry?

Reply:
Registration could fail if the script fails to download, parse, or throws an error in its initial execution, so, it should be successful unless there's something really wrong like syntax error or trying to access a non-existing resource in the cache, in which case, I don't think retrying would be of much use. If we decide to communicate this to the client, we could display a toast of something along the lines of "The web app cannot work offline if connection is lost" @delasous

Reply:
I think we should definitely toast the successful install so that the user knows the app will continue to work offline. But I'm not sure I see a real reason for them to know when it fails as they can't do much about it. Best thing for now would probably be to log the error with Sentry so that we're made aware of it. @silviot is this already setup/available for Librocco?